### DISCUSSION TITLE
Is the Power Apps be the only way to design form for list item on Sharepoint online?

### DISCUSSION THREAD

**Discussion by kenszeapp** [SharePoint/sp-dev-docs#8543] [Q&A] [answered]
I am a c# programmer, but new for SharePoint. Now, I am trying to Proof of concept that create an approval flow on SharePoint Online platform.  The target version is SharePoint online version, not on-premise. 

I find that the Power Apps is able to design form for list item on SharePoint online platform. Is it the only way?  What about C#, Asp.net or js? 

Moreover, is the Power automate look like the only way to handle application logic.  Is it ture?   

Thank you in advanced you read my post. 

![image](https://user-images.githubusercontent.com/25942774/199201347-dcbbc14d-8a5a-419f-8cb7-37d424dac5d1.png)

**Comment by AJIXuMuK** [comment, marked as answer]
@kenszeapp - you can use [SPFx Form Customizer Extension](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/building-form-customizer)
